packages/api/test/e2e/doughnut.e2e.ts

```diff
@@ -0,0 +1,206 @@
+        const opt = {doughnut: doughnutB};
+        await tx.signAndSend(holderKeyPair, opt as any, async ({events, status}) => {
+            if (status.isFinalized) {
+                const blockHash = status.value.toHex();
+                console.log('Completed at block hash', blockHash);
+                const allEvents = await api.query.system.events.at(blockHash);
+                const transfer = allEvents.find(
+                    event => (
+                        event.event.data.method === 'Transferred' &&
+                        event.event.data.section === 'genericAsset' &&
+                        event.event.data[1].toString() === alice.address && // transferred from alice's account
+                        event.event.data[0].toString() === CENNZ.toString() && // CENNZ
+                        event.event.data[2].toString() === reciever && // Dave is reciever
+                        event.event.data[3].toString() === amount.toString()
+                    ),
+                );
+                allEvents.forEach(({phase, event: {data, method, section}}) => {
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                });
+                if (transfer != undefined) {
+                    done();
+                } else {
+                    assert(true, 'false');
+                }
+            }
+        });
+    });
+
+    it('Fails when charlie uses bob\'s doughnut', async () => {
+        let doughnut = await makeDoughnut(
+            aliceKeyPair,
+            bob,
+            {'cennznet': makeCennznut('generic-asset', 'transfer')},
+        );
+        const doughnutB = new Doughnut(api.registry, doughnut);
+        const opt = {doughnut: doughnutB};
+        const tx = api.tx.genericAsset.transfer(16001, charlie.address, 10000);
+
+        await expect(tx.signAndSend(charlie, opt as any, () => {
+        })).rejects.toThrow('submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: {"Custom":180}');
+    });
+
+    it('fails when cennznut does not authorize the extrinsic method', async (done) => {
+        let doughnut = await makeDoughnut(
+            aliceKeyPair,
+            bob,
+            {'cennznet': makeCennznut('generic-asset', 'mint')},
+        );
+        const doughnutB = new Doughnut(api.registry, doughnut);
+        const opt = {doughnut: doughnutB};
+        const tx = api.tx.genericAsset.transfer(16001, charlie.address, 10000);
+
+        await tx.signAndSend(bob, opt as any, async ({events, status}) => {
+            if (status.isFinalized) {
+                const blockHash = status.value.toHex();
+                console.log('Completed at block hash', blockHash);
+                const allEvents = await api.query.system.events.at(blockHash);
+                const failed = allEvents.find(event => event.event.data.method === 'ExtrinsicFailed');
```

Comment:
The same as previous

Reply:
Is there any way to make sure the failure is because charlie's using bob's doughnut? What do we get in other fields of the extrinsic failed event?

Reply:
No.. cause the way node sends is.. {"Other":null} and extrinsic failed.. Error is Other

Reply:
I would then suggest in the same test, first try to use the created doughnut properly to prove everything is okay, then let the charlie use the same doughnut and show it fails.